src/policy/semantic.rs

```diff
@@ -25,15 +25,22 @@ use std::str::FromStr;
 
 /// Abstract policy which corresponds to the semantics of a Miniscript
 /// and which allows complex forms of analysis, e.g. filtering and
-/// normalization
+/// normalization.
+/// Semantic Policies don't have a `Pk` variant because they should have been
+/// converted to Pkh by `into_lift()` function.
+/// This is allow to following invariant.
+/// `Lift(Concrete) == Concrete -> Miniscript -> Script -> Miniscript -> Semantic`
+/// Lifting concrete policy is equiavalent to the following:
+/// Compiling Concrete to Miniscript followed by
+/// Encoding the miniscript to Script
+/// Decoding it back to Miniscript
+/// And finally lifting it back to Semantic
```

Comment:
I find this comment very confusing. It could be replaced by

```
Semantic policies store only hashes of keys to ensure that objects representing the same policy are lifted to the same `Semantic`, regardless of their choice of `pk` or `pk_h` nodes.
```